runtime/builtin_types.go

```diff
@@ -631,15 +631,39 @@ func builtinRound(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 			return nil, raised
 		}
 	}
-	// TODO: implement this.
-	if ndigits != 0 {
-		return nil, f.RaiseType(NotImplementedErrorType, "round with ndigits is not implemented in grumpy")
-	}
 	number, isFloat := floatCoerce(args[0])
+
 	if !isFloat {
 		return nil, f.RaiseType(TypeErrorType, "a float is required")
 	}
-	return NewFloat(math.Floor(number + 0.5)).ToObject(), nil
+
+	if math.IsNaN(number) || math.IsInf(number, 0) || number == 0.0 {
+		return NewFloat(number).ToObject(), nil
+	}
+
+	var result float64
+	if ndigits == 0 {
+		result = round(number, ndigits)
+		// round(12.5) => 13, round(-12.5) => -13
```

Comment:
My intuition is that negative ndigits should "just work" with the normal logic because we're using pow.

Additionally, my intuition about negative values for `number` is that we should just be able to do the same logic we do for positive values except on `-number`.

What I'm hoping for from a comment is to help readers understand why it's so much more complicated than that.